refinery/core/models.py

```diff
@@ -1873,11 +1880,8 @@ def get_managed_group(self):
             return None
 
     def save(self, *args, **kwargs):
-        if len(self.name) == 0:
-            logger.error("Group name cannot be empty.")
-            return
-        else:
-            super(ExtendedGroup, self).save(*args, **kwargs)
+        self.name = self.clean()
```

Comment:
The clean() method does not return a value (https://docs.djangoproject.com/en/1.7/_modules/django/db/models/base/#Model.clean). Also, validation should be done using the full_clean() method (https://docs.djangoproject.com/en/1.7/ref/models/instances/#django.db.models.Model.full_clean). Finally, overriding the save() method might cause the validation to be run twice in some cases and not run at all in others (http://stackoverflow.com/a/23684110).